framework/resource/retryresource/crud_resource.go

```diff
@@ -0,0 +1,230 @@
+package retryresource
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+
+	"github.com/giantswarm/operatorkit/framework"
+)
+
+const (
+	// Name is the identifier of the resource.
+	Name = "retry"
+)
+
+type crudResourceConfig struct {
+	Logger   micrologger.Logger
+	Resource framework.CRUDResourceOps
+
+	BackOff backoff.BackOff
+}
+
+type crudResource struct {
+	logger   micrologger.Logger
+	resource framework.CRUDResourceOps
```

Comment:
This file is a copy of previous `resource.go` with names changed and this line changed. It takes `resource framework.CRUDResourceOps` instead of `resource framework.Resource`.